src/dscheck/multicore_magic.ml

```diff
@@ -0,0 +1,44 @@
+module Atomic = Dscheck.TracedAtomic
+
+let copy_as_padded = Fun.id
+let make_padded_array = Array.make
+let length_of_padded_array = Array.length
+let length_of_padded_array_minus_1 xs = Array.length xs - 1
+
+module Transparent_atomic = struct
+  include Atomic
+
+  let make_contended = make
+  let fenceless_get = get
+  let fenceless_set = set
+end
+
+let[@inline] fenceless_get (atomic : 'a Atomic.t) =
+  !(Sys.opaque_identity (Obj.magic atomic : 'a ref))
+
+let[@inline] fenceless_set (atomic : 'a Atomic.t) value =
+  (Obj.magic atomic : 'a ref) := value
```

Comment:
These would actually actually cause problems, because `Atomic` refers to `Dscheck.TracedAtomic`.

So, we need to change these to:
```ocaml
let fenceless_get = Atomic.get
let fenceless_set = Atomic.set
```

Ideally we would support relaxed atomics in Dscheck at some point.